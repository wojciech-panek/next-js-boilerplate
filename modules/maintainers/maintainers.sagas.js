import { put, takeLatest } from 'redux-saga/effects';
import { delay } from 'redux-saga';

import reportError from '../../shared/utils/reportError';
import { MaintainersTypes, MaintainersActions } from './maintainers.redux';

const tmpData = [
  {
    firstName: 'Wojciech',
    lastName: 'Panek',
    email: 'wpanek@apptension.com',
  },
  {
    firstName: 'Patryk',
    lastName: 'Ziemkowski',
    email: 'pziemkowski@apptension.com',
  }
];

export function* fetchMaintainers() {
  try {
    yield delay(1000);

    return yield put(MaintainersActions.fetchSuccess(tmpData));
  } catch (e) {
    if (e.response) {
      return yield put(MaintainersActions.fetchError(e.response.data));
    }

    return yield reportError(e);
  }
}

export function* watchMaintainers() {
  try {
    yield takeLatest(MaintainersTypes.FETCH, fetchMaintainers);
  } catch (e) {
    yield reportError(e);
  }
}
