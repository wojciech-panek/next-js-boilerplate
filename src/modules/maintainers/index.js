export {
  MaintainersTypes,
  MaintainersActions,
} from './maintainers.redux';

export * from './maintainers.selectors';
