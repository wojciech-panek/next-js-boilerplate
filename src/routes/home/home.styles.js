import styled from 'styled-components';


export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
`;

export const Img = styled.img`
  width: 100px;
`;

