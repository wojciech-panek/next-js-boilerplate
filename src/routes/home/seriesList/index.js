export { default as SeriesList } from './seriesList.container';
