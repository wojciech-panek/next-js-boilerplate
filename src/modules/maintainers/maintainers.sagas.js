import { put, takeLatest } from 'redux-saga/effects';

import reportError from '../../shared/utils/reportError';
import api from '../../shared/services/api';
import { MaintainersTypes, MaintainersActions } from './maintainers.redux';


export function* fetchMaintainers() {
  try {
    const { data } = yield api.get('/static/mock-api/maintainers.json');

    return yield put(MaintainersActions.fetchSuccess(data));
  } catch (e) {
    if (e.response) {
      return yield put(MaintainersActions.fetchError(e.response.data));
    }

    return yield reportError(e);
  }
}

export function* watchMaintainers() {
  try {
    yield takeLatest(MaintainersTypes.FETCH, fetchMaintainers);
  } catch (e) {
    yield reportError(e);
  }
}
