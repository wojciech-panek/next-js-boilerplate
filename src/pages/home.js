import React, { PureComponent } from 'react';

import { Home } from '../routes/home';
import { MaintainersActions } from '../modules/maintainers/maintainers.redux';


export default class HomePage extends PureComponent {
  static async getInitialProps({ isServer, store }) {
    await store.execSagaTasks(isServer, (dispatch) => {
      dispatch(MaintainersActions.fetch());
    });

    return {};
  }

  render() {
    return <Home />;
  }
}
