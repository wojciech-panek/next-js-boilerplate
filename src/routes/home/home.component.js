import React, { PureComponent } from 'react';
import Head from 'next/head';
import { FormattedMessage } from 'react-intl';

import LogoPNG from '../../images/logo.png';
import LogoSVG from '../../images/icons/logo.svg';
import { H1 } from '../../shared/theme/typography';
import { Container, IconsContainer, Img } from './home.styles';
import messages from './home.messages';


export class Home extends PureComponent {
  static propTypes = {

  };

  render() {
    return (
      <Container>
        <FormattedMessage {...messages.pageTitle}>
          {pageTitle => (
            <Head>
              <title key="title">{ pageTitle }</title>
            </Head>
          )}
        </FormattedMessage>

        <H1>
          <FormattedMessage {...messages.welcome} />
        </H1>

        <IconsContainer>

          Example images usage:
          <Img src={LogoPNG} alt="" />
          <LogoSVG style={{ width: '25px' }} />
        </IconsContainer>
      </Container>
    );
  }
}
