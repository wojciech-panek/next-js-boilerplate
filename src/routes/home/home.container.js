import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'ramda';
import { withRouter } from 'next/router';

import withIntl from '../../shared/lib/withIntl';
import { Home } from './home.component';
import { selectLocalesLanguage } from '../../modules/locales/locales.selectors';
import { selectMaintainersItems } from '../../modules/maintainers/maintainers.selectors';


const mapStateToProps = createStructuredSelector({
  language: selectLocalesLanguage,
  maintainers: selectMaintainersItems,
});

export const mapDispatchToProps = (dispatch) => bindActionCreators({
}, dispatch);

export default compose(
  withIntl,
  withRouter,
  connect(mapStateToProps, mapDispatchToProps),
)(Home);
