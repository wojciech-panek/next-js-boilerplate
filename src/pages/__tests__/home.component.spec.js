import React from 'react';
import { shallow } from 'enzyme';
import { spy } from 'sinon';
import { expect } from 'chai';

import { HomePage } from '../home';
import { MaintainersActions } from '../../modules/maintainers/maintainers.redux';


describe('Home: Page', () => {
  const defaultProps = {
  };

  const component = (props) => (
    <HomePage {...defaultProps} {...props} />
  );

  it('should render correctly', () => {
    const wrapper = shallow(component());
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should dispatch MaintainersActions.fetch in getInitialProps', async () => {
    const dispatch = spy();
    const store = global.utils.prepareInitialPropsStore(dispatch);

    await HomePage.getInitialProps({ store });

    expect(dispatch).to.have.been.calledOnce;
    expect(dispatch).to.have.been.calledWith(MaintainersActions.fetch());
  });
});
