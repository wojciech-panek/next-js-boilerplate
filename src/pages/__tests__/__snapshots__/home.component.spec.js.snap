// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home: Page should render correctly 1`] = `<Connect(Home) />`;
